src/main/webapp/app/exam/participate/exercises/programming/programming-exam-submission.component.ts

```diff
@@ -34,28 +46,64 @@ export class ProgrammingExamSubmissionComponent extends ExamSubmissionComponent
         return this.exercise;
     }
 
-    @ViewChild(ExamCodeEditorStudentContainerComponent, { static: false })
-    codeEditorComponent: ExamCodeEditorStudentContainerComponent;
-
     isSaving: boolean;
     readonly ButtonType = ButtonType;
     readonly ButtonSize = ButtonSize;
 
+    /**
+     * On init set up the route param subscription.
+     * Will load the participation according to participation Id with the latest result and result details.
+     */
+    ngOnInit(): void {
+        // We lock the repository when the buildAndTestAfterDueDate is set and the due date has passed.
+        const dueDateHasPassed = !this.exercise.dueDate || moment(this.exercise.dueDate).isBefore(moment());
+        this.repositoryIsLocked = !!this.exercise.buildAndTestStudentSubmissionsAfterDueDate && !!this.exercise.dueDate && dueDateHasPassed;
+
+        const participation = { ...this.studentParticipation, exercise: this.exercise } as StudentParticipation;
+        this.domainService.setDomain([DomainType.PARTICIPATION, participation]);
+    }
+
+    reload(): void {
```

Comment:
Is there a reason why you keep this method?

Reply:
This was related to the issue mentioned by @kloessst . I uncommented these lines now.